plugins/crd/plugin_impl_crd.go

```diff
@@ -243,51 +245,213 @@ func createCRD(plugin *Plugin, FullName, Group, Version, Plural, Name string) er
 }
 
 func informerNetworkServices(plugin *Plugin) {
-	plugin.informerNS = plugin.sharedFactory.Networkservice().V1().NetworkServices().Informer()
+	// We use a shared informer from the informer factory, to save calls to the
+	// API as we grow our application and so state is consistent between our
+	// control loops. We set a resync period of 30 seconds, in case any
+	// create/replace/update/delete operations are missed when watching
+	plugin.sharedFactoryNS = factory.NewSharedInformerFactory(plugin.crdClient, time.Second*30)
+
+	plugin.informerNS = plugin.sharedFactoryNS.Networkservice().V1().NetworkServices().Informer()
 	// We add a new event handler, watching for changes to API resources.
 	plugin.informerNS.AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc: networkserviceEnqueue,
+			AddFunc: func(obj interface{}) {
+				var newEvent handler.NsmEvent
+				var err error
+				newEvent.Key, err = cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
```

Comment:
`cache.MetaNamespaceKeyFunc` should be used instead

Reply:
Done